src/validation/joi/joi.shared.schemas.test.ts

```diff
@@ -218,3 +219,35 @@ describe('uuidSchema', () => {
     expect(() => validate('123g4567-e89b-12d3-a456-426614174000', uuidSchema)).toThrow()
   })
 })
+
+describe('dateTimeStringSchema', () => {
+  const validDateTimes = [
+    '2024-09-30T00:55',
+    '2024-09-30T00:55+02:00',
+    '2024-09-30T00:55Z',
+    '2024-09-30T00:55:12',
+    '2024-09-30T00:55:12+02:00',
+    '2024-09-30T00:55:12Z',
+  ]
+
+  test.each(validDateTimes)('valid dateTime: %s', s => {
+    expect(validate(s, dateTimeStringSchema)).toBe(s)
+  })
+
+  const invalidDateTimes = [
+    undefined, // Non-string
+    null, // Non-string
+    '', // Empty string
+    '2024-07-25', // Missing time part
+    'T00:55:12Z', // Missing date
+    '2024-07-25T0055Z', // Missing colon in time
+    '2024-07-25 00:55', // Missing "T" between date and time
+    '2024/07/25T00:55', // Invalid date separator ("/" instead of "-")
+    '2024-07-25T00-55', // Invalid time separator ("-" instead of ":")
+    '2024-07-25T00:55Zextra', // Extra characters after a valid datetime
```

Comment:
I'd add more test cases to be extra sure:
```suggestion
    '2024-07-25T00:55Zextra', // Extra characters after a valid datetime
    'extra2024-07-25T00:55Z', // Extra characters before a valid datetime
    'Some random string', // Random string
    '2024 was a good year', // Year with some text
```